operator-common/src/main/java/io/strimzi/operator/common/controller/ControllerQueue.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.controller;
+
+import io.strimzi.operator.common.metrics.ControllerMetricsHolder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * Controller queue class wraps a Blocking queue and exposes the methods used by controllers. This includes taking
+ * events from the queue and enqueueing events into the queue.
+ */
+public class ControllerQueue {
+    private final static Logger LOGGER = LogManager.getLogger(ControllerQueue.class);
+
+    /*test*/ final BlockingQueue<SimplifiedReconciliation> queue;
+    private final ControllerMetricsHolder metrics;
+
+    /**
+     * Creates the controller queue. The controller should normally exist once per operator for cluster-wide mode or once per
+     * namespace for namespaced mode.
+     *
+     * @param queueSize     The capacity of the work queue
+     * @param metrics       Holder for the controller metrics
+     */
+    public ControllerQueue(int queueSize, ControllerMetricsHolder metrics) {
+        this.queue = new ArrayBlockingQueue<>(queueSize);
+        this.metrics = metrics;
+    }
+
+    public SimplifiedReconciliation take() throws InterruptedException {
+        return queue.take();
+    }
+
+    /**
+     * Enqueues the next reconciliation. It checks whether another reconciliation for the same resource is already in
+     * the queue and enqueues the new event only if it is not there yet.
+     *
+     * @param reconciliation    Reconciliation identifier
+     */
+    public void enqueue(SimplifiedReconciliation reconciliation)    {
+        if (!queue.contains(reconciliation)) {
```

Comment:
Yes, I think we do. It helps to avoid unnecessary reconciliations and makes sure the queue has an upper bound limit => it cannot have more items than we have KafkaUsers. 

Keep in mind that we are watching both the KafkaUser resource and Secrets. So a newly created KafkaUser resource would trigger 3reconciliations (1 when the KafkaUser is added, 1 when the Secret is created and 1 when the status is updated in the KafkaUser resource). So there will be duplicates in the queue if we don't check it.